packages/yarnpkg-pnpify/sources/buildNodeModulesTree.ts

```diff
@@ -110,6 +111,14 @@ export const buildLocatorMap = (nodeModulesTree: NodeModulesTree): NodeModulesLo
   return map;
 };
 
+function isPortalLocator(locatorKey: LocatorKey): boolean {
+  let descriptor = structUtils.parseDescriptor(locatorKey);
+  if (structUtils.isVirtualDescriptor(descriptor))
+    descriptor = structUtils.devirtualizeDescriptor(descriptor);
+
+  return descriptor.range.startsWith('portal:');
+};
```

Comment:
Side note: I feel like the nm linker makes a lot of similar assumptions ("starts with portal:", "starts with link:", etc). If possible (seems to be the easiest way in this particular instance, tbf) I think they should be kept to a minimum, as otherwise the linker becomes de facto monolithic, and replaced by semantic values (ie rather than hardcoding portals, what makes portals different from others? etc).

Just something to keep in mind for future developments 🙂

Reply:
I understand and agree. Portals are different because they almost all of the time point to the external directory, outside of project root. Nm linker should not write outside project root due to security concerns. I am just lazy and hadn't succeed from the first time to properly determine the target directory of portal dependencies, doing `portal:` check was easier. I will keep this technical debt in mind and will fix it at some point :)